compiler/src/dotty/tools/dotc/reporting/MessageKind.scala

```diff
@@ -0,0 +1,25 @@
+package dotty.tools.dotc.reporting
+
+/** The various message kinds that can be used inside of a Message
+ *  @param name What will be pontentially displayed to the user
+ */
+enum MessageKind(name: String):
+  case NoKind extends MessageKind("No Kind")
+  case Syntax extends MessageKind("Syntax")
+  case Type extends MessageKind("Type")
+  case TypeMismatch extends MessageKind("Type Mismatch")
```

Comment:
Almost tempted to use ``case `Type Mismatch` `` to reuse the default equals, but that might be too cute :).

Reply:
haha, it's funny you say that. I went back and forth _multiple_ times on this because it annoyed me that I needed to pass `name` into here, but once I realized that one of the names didn't follow the patter of `Each New Word Upper` like `Match case Unreachable` does, then I just caved and decided to pass in a name.

Reply:
It's not clear to me that there's a specific reason for "case" being lowercase here (also there is exactly one message of that kind, so maybe it could be folded into a more general case...)

Reply:
I thought since `case` was actually a keyword here we were just trying to be smart ™️ and use it lowercase.